pkg/rbac/rbac.go

```diff
@@ -145,6 +160,39 @@ const (
 	UserWrite = 1 << iota
 )
 
+// --
+// Required / Implied permissions.
+// Write permissions require subordinate read.
+// --
+
+var (
+	// requiredPermissions is not exported since maps cannot be constant.
+	requiredPermission = map[Permission][]Permission{
+		APIKeyWrite:    {APIKeyRead},
+		SettingsWrite:  {SettingsRead},
+		MobileAppWrite: {MobileAppRead},
+		UserWrite:      {UserRead},
+	}
+
+	// This is the inverse of the above map, set by the init() func.
+	// Done in code to ensure it always stays in sync with requiredPermission.
+	impliedBy = make(map[Permission][]Permission)
+)
+
+// Note: there are multiple init functions in this file. They are organized to be
```

Comment:
This is very clever and require/implied will always be in-sync.

but would it be easier to read as:
`requiredPermission = map[Permission][]Permission{
		APIKeyRead:    {APIKeyWrite},
		SettingsRead:  {SettingsWrite},
		MobileAppRead: {MobileAppWrite},
		UserRead:      {UserWrite},
	}`

and perhaps keep init() in but instead just compare them and `Panic`  if they don't match up. That would also allow for them to be asymmetrical eg. 
`APIKeyWrite:    {APIKeyRead, SuperPowers},`
and
`APIKeyRead:    {APIKeyWrite},`

Reply:
I realize I'm advocating both duplication and panic, but multiple-init is also weird

Reply:
Multi-init is pretty common tbh

Reply:
This actually allows for the asymmetry - we just don't have it yet.